src/module/item/flows/PackActionFlow.ts

```diff
@@ -0,0 +1,153 @@
+import { SR5Actor } from "@/module/actor/SR5Actor";
+import { SR5Item } from "../SR5Item";
+import { FLAGS, SYSTEM_NAME } from "@/module/constants";
+import { SR5 } from "@/module/config";
+import { Helpers } from "@/module/helpers";
+
+/**
+ * Handle interaction with the system packs containing predefined actions.
+ * 
+ * This includes packs like general and matrix actions.
+ */
+export const PackActionFlow = {
+    /**
+     * Return the matrix action pack name to use, when the matrix actions pack is referenced.
+     */
+    getMatrixActionsPackName(): Shadowrun.PackName {
+        const overrideMatrixPackName = game.settings.get(SYSTEM_NAME, FLAGS.MatrixActionsPack) as Shadowrun.PackName;
+        return overrideMatrixPackName || SR5.packNames.MatrixActionsPack as Shadowrun.PackName;
+    },
+
+    /**
+     * Returns the general actions pack name to use, when the general actions pack is referenced.
+     */
+    getGeneralActionsPackName(): Shadowrun.PackName {
+        const overrideGeneralpackName = game.settings.get(SYSTEM_NAME, FLAGS.GeneralActionsPack) as Shadowrun.PackName;
+        return overrideGeneralpackName || SR5.packNames.GeneralActionsPack as Shadowrun.PackName;
+    },
+
+    /**
+     * Return the matrix action pack name to use, when the matrix actions pack is referenced.
+     */
+    getICActionsPackName(): Shadowrun.PackName {
+        const overrideMatrixPackName = game.settings.get(SYSTEM_NAME, FLAGS.ICActionsPack) as Shadowrun.PackName;
+        return overrideMatrixPackName || SR5.packNames.ICActionsPack as Shadowrun.PackName;
+    },
+
+    /**
+     * Retrieve all actions from a given pack.
+     *
+     * Other item types in that pack will be ignored.
+     *
+     * TODO: Allow filtering by categories?
+     * TODO: Generalize this to search for items of a certain type?
+     *
+     * @param packName The item pack that contains actions.
+     */
+    async getPackActions(packName: string): Promise<SR5Item<'action'>[]> {
+        console.debug(`Shadowrun 5e | Trying to fetch all actions from pack ${packName}`);
+        const pack = game.packs.find(pack => pack.metadata.system === SYSTEM_NAME && pack.metadata.name === packName);
+        if (!pack) return [];
+
+        // @ts-expect-error foundry-vtt-types v10
+        const packEntries = pack.index.filter(data => data.type === 'action');
```

Comment:
the error here is that it doesn't know the type of the document you are getting, it can be Actor, Item, Scene, etc...
So, on pack, you can cast to ` as CompendiumCollection<'Item'> | undefined`, so it won't give the error in this line.